tests/unittests/test_handler/test_handler_ca_certs.py

```diff
@@ -201,6 +201,29 @@ def test_single_cert_no_trailing_cr(self):
 
             mock_load.assert_called_once_with("/etc/ca-certificates.conf")
 
+    def test_single_cert_to_empty_existing_ca_file(self):
+        """Test adding a single certificate to the trusted CAs
+        when existing ca-certificates.conf is empty"""
+        cert = "CERT1\nLINE2\nLINE3"
+
+        ca_certs_content = ""
+        expected = "cloud-init-ca-certs.crt\n"
+
+        with ExitStack() as mocks:
+            mock_write = mocks.enter_context(
+                mock.patch.object(util, 'write_file'))
+            mock_load = mocks.enter_context(
+                mock.patch.object(util, 'load_file',
+                                  return_value=ca_certs_content))
```

Comment:
Following roughly this pattern, I think mocking `os.stat` would look something like this:

```py
mock_stat = mocks.enter_context(
    mock.patch("cloudinit.config.cc_ca_certs.os.stat")
)
mock_stat.return_value.st_size = 0
```

(Note that passing a string to `mock.patch` means that it will perform the lookup for us and shadow the given symbol; for stdlib modules it's always risky to monkeypatch them directly because you can easily violate test framework/code under test invariants without knowing it.)